packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -991,17 +991,27 @@ class TextPainter {
     canvas.drawParagraph(_paragraph!, offset);
   }
 
-  // Returns true iff the given value is a valid UTF-16 high surrogate. The value
-  // must be a UTF-16 code unit, meaning it must be in the range 0x0000-0xFFFF.
-  //
-  // See also:
-  //   * https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF
-  static bool _isHighSurrogate(int value) {
+  /// Returns true iff the given value is a valid UTF-16 high (first) surrogate.
+  /// The value must be a UTF-16 code unit, meaning it must be in the range
+  /// 0x0000-0xFFFF.
+  ///
+  /// See also:
+  ///   * https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF
+  ///   * [isLowSurrogate], which checks the same thing for low (second)
+  /// surrogates.
+  static bool isHighSurrogate(int value) {
```

Comment:
nit: maybe assert `value` is in the correct range?

Reply:
I just tried this on your suggestion, but then realized that there were a million test failures.  I think these methods check lots of code units that are not necessarily surrogates.

Reply:
sorry I meant valid UTF16 code range 0x0 – 0xFFFFF

Reply:
Ohh agreed, I'll do that!